Updates troubleshooting page (#127)
NUbots/NUbook: src/book/03-guides/04-general/02-troubleshooting.mdx

If this page does not solve your problem, the following people can help you with specific parts of the system:

- **Hardware**: Taylor Young
- **Toolchain and build**: Kip Hamiltons
- **NUClear**: Cameron Murtagh
- **NUsight**: Josephus Paye II, Brendan Annable
- **Machine Learning**: Joel Wong
- **Mathematics**: Kip Hamiltons, Cameron Murtagh, Taylor Young, Ysobel Sims

> Team mentors Alex Biddulph and Trent Houliston can help with most parts of the system if this list does not help.

Reviewer: Might be missed if only looking at the list above

```suggestion
Team mentors **Alex Biddulph** and **Trent Houliston** can help with most parts of the system if this list does not help.
```

If the problem persists, try purging the build volume by running `./b configure --clean`.
Note that purging the build volume means you'll have to rerun `./b build` to use built code.

> ### Not enough memory in Docker? (Mac OS)

Reviewer: ```suggestion
### Not enough memory in Docker? (macOS)
```

> By default the docker install will allocate 2GB of memory to docker images. When building the images this is insufficent for some of the libraries and will result in an error. In the menu bar go to advanced settings and increase the memory available.

Reviewer: ```suggestion
By default Docker Desktop will allocate 2GB of memory to use for building images. When building the NUbots image, this is insufficient and will result in an error. To fix this, open the Docker menu in the menu bar, go to advanced settings, and increase the memory available.
```

## NUsight

### Multiple Docker robots in NUsight?

This means someone else is also running a Docker robot. You can modify the name of your Docker robot by doing the following.

> 
> ```bash
> nano module/support/configuration/NetworkConfiguration/data/config/NetworkConfiguration.yaml
> ```

Reviewer: ```suggestion

1. Open the `NetworkConfiguration.yaml` file in a text editor (here we are using nano on the command line):

   ```bash
   nano module/support/configuration/NetworkConfiguration/data/config/NetworkConfiguration.yaml
   ```
```

Author: ```suggestion

1. Open the `NetworkConfiguration.yaml` file in a text editor (here we are using nano on the command line):
   ```bash
   nano module/support/configuration/NetworkConfiguration/data/config/NetworkConfiguration.yaml
   ```
```

Author: Looks like the ending ``` got cut off, good job GitHub 😆 
I'll commit and add it in manually.

> Write in the name you want to use in the `name: ""` section and press `CTRL+S` to save the file and then `CTRL+X` to exit.

Reviewer: ```suggestion
2. Update the `name` field with the name you want to use and press <kbd>Ctrl</kbd>+<kbd>S</kbd> to save the file and then <kbd>Ctrl</kbd>+<kbd>X</kbd> to exit.
```

> Rebuild using `./b build` for the changes to take effect.

Reviewer: ```suggestion
3. Rebuild using `./b build` for the changes to take effect.
```

> In NUsight, toggle off the robots you do not want to see by clicking the robot icon in the top right of the NUsight GUI and toggling the slider next to the name of the robot you want to toggle off.

Reviewer: ```suggestion
4. In NUsight, toggle off the robots you do not want to see by clicking the robot icon at the top right of the NUsight GUI and toggling the slider next to the name of every other robot except the one you set in step 2.
```
